auth/plain.go

```diff
@@ -11,7 +22,15 @@ type Plain struct {
 
 func (p *Plain) Check(c server.ClientAuth) bool {
 	opts := c.GetOpts()
-	if p.Username != opts.Username || p.Password != opts.Password {
+	if p.Username != opts.Username {
+		return false
+	}
+	// Check to see if the password is a bcrypt hash
+	if isBcrypt(p.Password) {
+		if err := bcrypt.CompareHashAndPassword([]byte(p.Password), []byte(opts.Password)); err != nil {
+			return false
```

Comment:
True, but at this point we have no context since auth is pluggable. We could return more elaborate errors, but in the end the password did not match. ;)